src/sentry/static/sentry/app/views/groupMerged/mergedToolbar.jsx

```diff
@@ -4,56 +4,106 @@ import Reflux from 'reflux';
 
 import {t} from '../../locale';
 import GroupingStore from '../../stores/groupingStore';
-
+import ProjectActions from '../../actions/projectActions';
+import Button from '../../components/buttons/button';
 import LinkWithConfirmation from '../../components/linkWithConfirmation';
 import Toolbar from '../../components/toolbar';
 import SpreadLayout from '../../components/spreadLayout';
 
 const MergedToolbar = React.createClass({
   propTypes: {
     onUnmerge: PropTypes.func,
-    onCollapse: PropTypes.func
+    groupId: PropTypes.string,
+    onToggleCollapse: PropTypes.func
   },
 
   mixins: [Reflux.listenTo(GroupingStore, 'onGroupingUpdate')],
+
   getInitialState() {
+    let {
+      unmergeList,
+      unmergeLastCollapsed,
+      unmergeDisabled,
+      enableFingerprintCompare
+    } = GroupingStore;
+
     return {
-      unmergeCount: 0
+      enableFingerprintCompare,
+      unmergeList,
+      unmergeLastCollapsed,
+      unmergeDisabled
     };
   },
 
-  onGroupingUpdate({unmergeList}) {
-    if (unmergeList && unmergeList.size !== this.state.unmergedCount) {
-      this.setState({
-        unmergeCount: unmergeList.size
-      });
-    }
+  onGroupingUpdate(updateObj) {
+    let allowedKeys = [
+      'unmergeLastCollapsed',
+      'unmergeDisabled',
+      'unmergeList',
+      'enableFingerprintCompare'
+    ];
+
+    Object.keys(updateObj).filter(key => allowedKeys.indexOf(key) > -1).forEach(key => {
+      if (typeof updateObj[key] !== 'undefined') {
```

Comment:
this could have been done (subjectively) more simply  with a filter on the lodash updateObj and a merge imo, but it's up to you and this looks correct